layouts/partials/search-results.html

```diff
@@ -11,15 +29,51 @@
           "url": "{{ .Permalink }}"
       },
       {{ end }}
-  }
+    }
 
+    const idx = lunr(function () {
```

Comment:
FWIW it appears this code was heavily inspired by https://victoria.dev/blog/add-search-to-hugo-static-sites-with-lunr/#search-for-the-query-with-lunr